test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -4334,6 +4476,274 @@ def test_cgroup_default_config(self):
+            # None will be seen as a failure, nothing to do
+            pass
+        self.logger.info("total available mem: %d"
+                         % mem_avail_in_bytes)
+        self.assertTrue(mem_avail_in_bytes is not None,
+                        "Unable to read total memsw available")
+
+        # Get total phys+swap memory available
+        vmem_avail = os.path.join(mem_base,
+                                  'memory.memsw.limit_in_bytes')
+        result = self.du.cat(hostname=self.mom.hostname, filename=vmem_avail,
+                             sudo=True)
+        vmem_avail_in_bytes = None
+        try:
+            vmem_avail_in_bytes = int(result['out'][0])
+        except Exception:
+            # None will be seen as a failure, nothing to do
+            pass
+        self.logger.info("total available memsw: %d"
+                         % vmem_avail_in_bytes)
+        self.assertTrue(vmem_avail_in_bytes is not None,
+                        "Unable to read total memsw available")
+
+        # Get job physical mem limit
+        mem_limit = os.path.join(mem_base, str(jid),
+                                 'memory.limit_in_bytes')
+        result = self.du.cat(hostname=self.mom.hostname, filename=mem_limit,
+                             sudo=True)
+        mem_limit_in_bytes = None
+        try:
+            mem_limit_in_bytes = int(result['out'][0])
+        except Exception:
+            # None will be seen as a failure, nothing to do
+            pass
+        self.logger.info("job mem limit: %d"
+                         % mem_limit_in_bytes)
+        self.assertTrue(mem_limit_in_bytes is not None,
+                        "Unable to read job mem limit")
+
+        # Get job phys+swap mem limit
+        vmem_limit = os.path.join(mem_base, str(jid),
+                                  'memory.memsw.limit_in_bytes')
+        result = self.du.cat(hostname=self.mom.hostname, filename=vmem_limit,
+                             sudo=True)
+        vmem_limit_in_bytes = None
+        try:
+            vmem_limit_in_bytes = int(result['out'][0])
+        except Exception:
+            # None will be seen as a failure, nothing to do
+            pass
+        self.logger.info("job memsw limit: %d"
+                         % vmem_limit_in_bytes)
+        self.assertTrue(vmem_limit_in_bytes is not None,
+                        "Unable to read job memsw limit")
+
+        # Check results correspond to enforcement flags and job placement
+        swap_avail = vmem_avail_in_bytes - mem_avail_in_bytes
+        if enforce_flags[0] == 'true' and not exclhost:
+            self.assertTrue(mem_limit_in_bytes == 100 * 1024 * 1024,
+                            "Job mem limit is %d expected %d"
+                            % (mem_limit_in_bytes, 100 * 1024 * 1024))
+        else:
+            self.assertTrue(mem_avail_in_bytes == mem_limit_in_bytes,
+                            "job mem limit is not total mem available")
```

Comment:
Can this message be more like the one on line 4717 and print the expected and got values?

Reply:
Sure.